README.md

```diff
@@ -0,0 +1,34 @@
+# n64chain
+
+This is a Nintendo 64 development toolchain based on GCC that do not depend on any proprietary Nintendo library.
+Only thing that you need to provide is a 2kb bootcode compatible with your CIC to bootstrap the code.
+This is not included in n64chain.
+
+The library includes basic OS functionality supporting multi-threading through hardware interrupts on the
+single MIPS 4300 CPU of the N64. Mainly written in C and MIPS assembly, it is expected to provide a
+lightweight development environment and thus requires some low-level baby sitting.
+
+## Building
+
+To build the toolchain, first put your boot code named `header.bin` on the `libn64` folder. Run `build-linux64-toolchain.sh`
+in the `tools` folder on a bash-compatible shell to start building the cross-compiler.
+Prerequisites are MPFR and MPC with development headers, which can be installed via `apt install libmpfr-dev libmpc-dev`
```

Comment:
`build-essential` (Debian/Ubuntu) or whatever the Redhat equivalent is.
Also, `libgmp-dev`, `yacc`, and `bison`.